src/opendkim.h

```diff
@@ -1,6 +1,10 @@
 #ifndef LIBOPENDKIM_H
 #define LIBOPENDKIM_H
 
+#if __GNUC__ > 4
+#define _Bool bool
+#endif
+
```

Comment:
This was tricky to track down.  In my development and production environments I am on `gcc` `4.x`, but in the docker container `gcc` was `6.x`.  This caused the following errors:
```
included from ../src/opendkim.h:11:0,
                 from ../src/opendkim.cc:1:
/usr/include/opendkim/dkim.h:456:30: error: '_Bool' has not been declared
 extern DKIM_STAT dkim_resign __P((DKIM *news, DKIM *olds, _Bool hdrbind));
[...]
```
I chased this down to a change in `stdbool.h` between versions and then made this macro.  I'm a little worried it was removed for a reason, but once I made this change, things compiled fine and the tests ran.  I'm open to suggestions on a better or more robust way to solve this.

Reply:
I think the "right" way to solve this is for dkim.h to only include `<stdbool.h>` if it's being compiled as C (not C++).  It looks like it has the `#ifdef __cplusplus; extern "C" { ...` trick, but I guess `stdbool.h` is missing checks to make it a no-op when compiled as C++, so dkim.h should do it instead.

Not sure there's anything you can do, and your solution seems fine to me!